src/SystemLog/Logger.cs

```diff
@@ -39,6 +40,9 @@ void ILogPlugin.Log(string source, LogLevel level, string message)
 
                 if (!string.IsNullOrEmpty(Settings.Default.Path))
                 {
+                    StringBuilder sb = new StringBuilder(source);
+                    foreach (char c in GetInvalidFileNameChars())
+                        sb.Replace(c, '-');
```

Comment:
Throw exception or fix in neo-core? I think we shouldn't modify the `source`, which is the log file name. 🤔

Reply:
I think that we should merge both

Reply:
I think we should only merge this one. `source` is not filename. We use it as filename in SystemLog, so we fix it in SystemLog.

Reply:
The problem is, user want a log file named `A`, but we create `B`, this is unreasonable.

Reply:
As I have said, the parameter `source` is not a filename. It is the log source. We just use it as filename in SystemLog.

Reply:
Got.